packages/perseus/src/components/math-input.tsx

```diff
@@ -240,36 +154,95 @@ class MathInput extends React.Component<Props, State> {
         if (_(value).isFunction()) {
             value(input);
         } else if (value[0] === "\\") {
-            input.cmd(value).focus();
+            input?.cmd(value).focus();
         } else {
-            input.write(value).focus();
+            input?.write(value).focus();
         }
-        input.focus();
+        input?.focus();
     };
 
-    mathField: (arg1: any) => any = (options) => {
-        // The MathQuill API is now "versioned" through its own "InterVer"
-        // system.
-        // See: https://github.com/mathquill/mathquill/pull/459
-        const MQ = MathQuill.getInterface(2);
-
-        // MathQuill.MathField takes a DOM node, MathQuill-ifies it if it's
-        // seeing that node for the first time, then returns the associated
-        // MathQuill object for that node. It is stable - will always return
-        // the same object when called on the same DOM node.
-        // eslint-disable-next-line react/no-string-refs
-        return MQ.MathField(ReactDOM.findDOMNode(this.refs.mathinput), options);
+    mathField: () => MathFieldInterface | null = () => {
+        if (!this.__mathField && this.__mathFieldWrapperRef) {
+            // Initialize MathQuill.MathField instance
+            this.__mathField = createMathField(
+                this.__mathFieldWrapperRef,
+                (baseConfig) => {
+                    return {
```

Comment:
A tiny trick I learned. You can simplify this slightly when the only point of a function is to return a value immediately. 
```suggestion
                (baseConfig) => ({
                
                })
```